client/src/ts/trainees/InternalProject.tsx

```diff
@@ -0,0 +1,330 @@
+import Paper from "@material-ui/core/Paper";
+import Button from "@material-ui/core/Button";
+import { makeStyles } from "@material-ui/core/styles";
+import { Typography, IconButton } from "@material-ui/core";
+import { Edit } from "@material-ui/icons";
+import * as traineesTypes from "./traineesTypes";
+import { InternalProjectInformationDialog } from "./InternalProjectInformationDialog";
+import { doesPermissionsHaveSomeOf } from "../utils/authUtils";
+import { InternalProjectDetail } from "./InternalProjectDetail";
+import { AuthContext } from "../global/AuthContext";
+import api from "../utils/api";
+import { showErrorMessage, showSuccessMessage } from "../utils/toastUtils";
+import { authReducerActionType } from "../global/globalTypes";
+
+/**
+ * Function which provides the styles
+ */
+const useStyles = makeStyles((theme) => ({
+  submit: {
+    margin: theme.spacing(3, 0, 1),
+    color: "white",
+  },
+  paper: {
+    margin: theme.spacing(3),
+    padding: theme.spacing(3),
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+  },
+  category: {
+    color: theme.palette.text.secondary,
+    width: "100%",
+  },
+  categoryHeader: {
+    display: "flex",
+    width: "100%",
+    justifyContent: "space-between",
+    alignItems: "center",
+  },
+
+  categoryTitle: {
+    textAlign: "center",
+  },
+  categoryLine: {
+    paddingTop: "12.5px",
+    paddingBottom: "11.5px",
+    textAlign: "right",
+  },
+
+  categoryItem: {
+    display: "flex",
+    justifyContent: "space-between",
+    [theme.breakpoints.up("md")]: {
+      width: "30%",
+    },
+    [theme.breakpoints.down("md")]: {
+      width: "50%",
+    },
+    [theme.breakpoints.down("sm")]: {
+      width: "60%",
+    },
+    [theme.breakpoints.down("xs")]: {
+      width: "100%",
+    },
+  },
+  categoryItemList: {
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "right",
+  },
+
+  button: {
+    padding: "6px 12px",
+    textAlign: "center",
+    textDecoration: "none",
+    display: "inline-block",
+    margin: "4px 2px",
+    cursor: "pointer",
+    backgroundColor: "white",
+    color: "black",
+    border: "none",
+  },
+  fullWidth: {
+    width: "100%",
+  },
+
+  submitContainer: {
+    width: "100%",
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "flex-end",
+  },
+
+  cancelButton: {
+    margin: theme.spacing(1, 1, 1, 1),
+  },
+  submitButton: {
+    margin: theme.spacing(1, 0, 1, 1),
+    color: "white",
+  },
+}));
+
+/**
+ * Interface for the match parameter of the router
+ */
+interface RouterMatch {
+  id: string;
+}
+
+/**
+ * Function that allows displays internal project details
+ * @returns returns the interface for the user
+ */
+const InternalProject: React.FunctionComponent<RouteComponentProps<RouterMatch>> = (
+  props: RouteComponentProps<RouterMatch>
+) => {
+  const classes = useStyles();
+
+  const { auth, dispatchAuth } = useContext(AuthContext);
```

Comment:
dispatchAuth is not used.
```suggestion
  const { auth } = useContext(AuthContext);
```